roles/gitlab-postconfig/tasks/main.yml

```diff
@@ -26,56 +26,127 @@
 #    " | gitlab-rails console production
 
 - name: GitLab Post | copy root password script to host
-  template: 
+  template:
     src: set_root_pw.sh.j2
     dest: /root/set_root_pw.sh
     owner: root
     group: root
     mode: 0700
 
-- name: GitLab Post | Execute set root password 
+- name: GitLab Post | Execute set root password
   command: ./set_root_pw.sh
-  args: 
+  args:
     chdir: /root
 
-- name: GitLab Post | Remove set root password script 
+- name: GitLab Post | Remove set root password script
   file:
     path: /root/set_root_pw.sh
     state: absent
 
 - name: sleep 2 minute to allow root to initialize after password set
-  pause: 
+  pause:
     minutes: 2
 
 ### Gitlab workaround
 ### Users are not populated from AD until login from Web UI 
 ### I haven't found an automated workaround... 
 
-- name: GitLab Post | Create local users
-  gitlab_user:
-    server_url: https://gitlab.{{ dns_domain_name }}
-    validate_certs: False
-    login_user: root
-    login_password: "{{ domain_admin_password }}"
-    name: "{{ user_prefix }}{{ item }}"
-    username: "{{ user_prefix }}{{ item }}"
-    password: "{{ users_password }}"
-    email: "{{ user_prefix }}{{ item }}@{{ dns_domain_name }}"
-    confirm: False
-    state: present
-  with_sequence: count="{{ user_count }}"
+- block:
 
-- name: GitLab Post | Create Projects in GitLab
-  gitlab_project:
-    server_url: https://gitlab.{{ dns_domain_name }}
-    validate_certs: False
-    login_user: root
-    login_password: "{{ domain_admin_password }}"
-    name: "{{ user_prefix }}{{ item }}"
-    group: "{{ user_prefix }}{{ item }}"
-    issues_enabled: False
-    wiki_enabled: False
-    snippets_enabled: False
-    state: present
-  with_sequence: count="{{ user_count }}"
-  ignore_errors: true
+    - name: GitLab Post | Create local users
+      gitlab_user:
+        server_url: https://gitlab.{{ dns_domain_name }}
+        validate_certs: False
+        login_user: root
+        login_password: "{{ domain_admin_password }}"
+        name: "{{ user_prefix }}{{ item }}"
+        username: "{{ user_prefix }}{{ item }}"
+        password: "{{ users_password }}"
+        email: "{{ user_prefix }}{{ item }}@{{ dns_domain_name }}"
+        confirm: False
+        state: present
+      with_sequence: count="{{ user_count }}"
+
+    - name: GitLab Post | Create Projects in GitLab
+      gitlab_project:
+        server_url: https://gitlab.{{ dns_domain_name }}
+        validate_certs: False
+        login_user: root
+        login_password: "{{ domain_admin_password }}"
+        name: "{{ user_prefix }}{{ item }}"
+        group: "{{ user_prefix }}{{ item }}"
+        issues_enabled: False
+        wiki_enabled: False
+        snippets_enabled: False
+        state: present
+      with_sequence: count="{{ user_count }}"
+      ignore_errors: true
+
+  when: ansible_version.full is version_compare('2.8', '<')
```

Comment:
For this, you might try upping the Gitlab version (if ansible 2.8 or beyond).  We are currently pinning the Gitlab version to a very old version because the Ansible modules before 2.8 didn't support the new API.  2.8 works with the latest API, so you might try changing the version to the latest to see if it works.

Reply:
Good thinking. I did try this and needed to upgrade the gitlab role to the latest to be able to install which I did and installed the latest version of gitlab successfully. However even after that installing the latest version of gitlab getting an error during the step where it uses gitlab_user and gitlab_project modules on ansible 2.8. Not sure what the real cause is. It seems to work ok on 2.7 albeit older version of gitlab. I added steps to use the uri module when using 2.8 and left the current implementation for <2.8. I will commit my changes which will add to this PR (new gitlab role as well as additional tweaks to allow to install the latest version). Let's test these changes on oa-develop branch and merge. We can rebase develop onto master after that. Going forward, I will commit against develop and do a PR from there.